### ISSUE TITLE
sbt 1.1.0 considers itself a member of the sbt 1.0 universe

### ISSUE THREAD

**Issue by paulp** [sbt/sbt#3858]
## problem

- sbt 0.12 looks in ~/.sbt/0.12
- sbt 0.13 looks in ~/.sbt/0.13
- sbt 1.0 looks in ~/.sbt/1.0
- sbt 1.1 still looks in ~/.sbt/1.0

## expectation

Consistency. Previously, changes in the minor version had a meaning which sbt 1.1.0 is violating. As best I can tell from the release notes this is intentional, but if it is intentional, it is not explained or motivated. And I'm not sure it's intentional.

## notes

sbt version: 1.1.0

**Comment by eed3si9n** [comment]
It's intentional. sbt 1.x adopts Semantic Versioning.

The English word binary is confusing because it means both "of two thing" and "your JAR is swappable". Until 1.1, both meanings lined up because there was a convention of using the first two segments of the full version number to signify binary-as-in-swap-JAR compatibility.

Because of Semantic Versioning, bincompat is now denoted only by the first segment. We should've caught this early on and used `1` or something, but there were some code expecting `(Int, Int)`, so we went with `1.0`.

Why? Keeping 1.0 prevents the plugin ecosystem from burning to the ground.

Why SemVar? I guess because I don't think we need to copy the second-segment convention, where the first segment means "epic." The whole time during 0.13 we were able to add features while sort of keeping the bincompat, so we are using second segment to mean new features.

Lately libraries like Akka and Cats are also adopting SemVer.

**Comment by dwijnand** [comment]
As Eugene said, 1.0 is the binary API for the whole 1.x series. In retrospect I should've made it "1.x" (and properly dealt with the existing (Int, Int) type signature) instead of "1.0". My bad.

I'm happy to keep this issue open to see how we can better communicate this now.

**Comment by smarter** [comment]
Surely it's not too late to make sbt understand "1.x" to mean exactly the same thing as "1.0" ? I'm afraid no amount of documentation is going to prevent this from confusing people until the end of times.

**Comment by mosesn** [comment]
I agree, this is really confusing.  I spent a while experimenting to try to publish a 1.1 version for https://github.com/twitter/scrooge/pull/291 until I was pointed here by a [stackoverflow answer][0].  If you really think it's too late for 1.x, let's at least support 2.x going forward.

[0]: https://stackoverflow.com/a/50083145/284648

**Comment by eed3si9n** [comment]
This was fixed in https://github.com/sbt/librarymanagement/pull/437 for sbt 2.x series.
